src/conn.c

```diff
@@ -3180,6 +3183,8 @@ natsConnection_ConnectTo(natsConnection **newConn, const char *url)
     s = natsOptions_Create(&opts);
     if (s == NATS_OK)
         s = _processUrlString(opts, url);
+    if (s != NATS_OK)
```

Comment:
If we make `_processUrlString()` fail, I would change to this to make it a bit more clear why we need to destroy the options in that case:
```
if (s == NATS_OK)
{
    s = _processUrlString(opts, url);
    // Until natsConn_create() takes ownership of the options, we are 
    // responsible for destroying it in case of error.
    if (s != NATS_OK)
        natsOptions_Destroy(opts);
}
```